pkg/apis/core/v1alpha1/uibutton_types.go

```diff
@@ -56,8 +56,26 @@ type UIButtonSpec struct {
 	// Location associates the button with another component for layout.
 	Location UIComponentLocation `json:"location" protobuf:"bytes,1,opt,name=location"`
 
-	// Text to appear on the button itself.
+	// Text to appear on the button itself or as hover text (depending on button location).
 	Text string `json:"text" protobuf:"bytes,2,opt,name=text"`
+
+	// IconName is a Material Icon to appear next to button text or on the button itself (depending on button location).
+	//
+	// Valid values are icon font ligature names from the Material Icons set.
+	// See https://fonts.google.com/icons for the full list of available icons.
+	//
+	// If both IconSVG and IconName are specified, IconSVG will take precedence.
+	//
+	// +optional
+	IconName string `json:"iconName,omitempty" protobuf:"bytes,3,opt,name=iconName"`
+
+	// IconSVG is an SVG to use as the icon to appear next to button text or on the button itself (dependong on button
```

Comment:
```suggestion
	// IconSVG is an SVG to use as the icon to appear next to button text or on the button itself (depending on button
```

Reply:
It seems like it'd be tempting for a user to put a path to an SVG file here. It'd be good to specify in the comment that this wants raw xml not a filename and maybe also validate that it's XML in `Validate` to catch people who miss the note in the docs?

Reply:
Good points - clarified in docs and added ridiculously basic validation (`strings.Contains(in.Spec.IconSVG, "<svg")`) + fixed that typo in a6d0254e8 (needed to re-run codegen so didn't accept GitHub suggestion 🤪).